src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -972,7 +972,7 @@ void ImageReplayer<I>::handle_shut_down(int r) {
       m_status_removed = true;
       handle_shut_down(r);
     });
-    remove_image_status(m_finished, ctx);
+    remove_image_status(m_finished || m_last_r == -EREMOTEIO, ctx);
```

Comment:
I am not sure it is quite right. `-EREMOTEIO` means that the local image is primary, i.e. it should be valid for it to have mirror status.

So even it does the right thing in some particular case I don't think it does it right in all cased.

Reply:
Wait but I think that `-EREMOTEIO` means that both images are non-primary actually. This seems to happen only with weird test/bad restart timing.

Reply:
Yes, it returns when the remote image is non-primary buit may seem also to be returned when local image is primary. In any case, I don't see reasons to remove mirror status in this case.